landingai/visualize.py

```diff
@@ -241,5 +238,61 @@ def _get_pil_font(font_size: int = 18) -> ImageFont.FreeTypeFont:
     ObjectDetectionPrediction: overlay_bboxes,
     SegmentationPrediction: overlay_colored_masks,
     ClassificationPrediction: overlay_predicted_class,
-    OcrPrediction: overlay_quadrilateral,
+    OcrPrediction: overlay_ocr_predictions,
 }
+
+
+def _draw_box_text(
+    img_size: Tuple[int, int], box: List[Tuple[int, int]], text: Optional[str] = None
+) -> np.ndarray:
+    box_height = int(
+        math.sqrt((box[0][0] - box[3][0]) ** 2 + (box[0][1] - box[3][1]) ** 2)
+    )
+    box_width = int(
+        math.sqrt((box[0][0] - box[1][0]) ** 2 + (box[0][1] - box[1][1]) ** 2)
+    )
+
+    if box_height > 2 * box_width and box_height > 30:
+        img_text = Image.new("RGB", (box_height, box_width), (255, 255, 255))
+        draw_text = ImageDraw.Draw(img_text)
+        if text:
+            font = _create_font(text, (box_height, box_width))
+            draw_text.text((0, 0), text, fill=(0, 0, 0), font=font)
+        img_text = img_text.transpose(Image.ROTATE_270)
+    else:
+        img_text = Image.new("RGB", (box_width, box_height), (255, 255, 255))
+        draw_text = ImageDraw.Draw(img_text)
+        if text:
+            font = _create_font(text, (box_width, box_height))
+            draw_text.text((0, 0), text, fill=(0, 0, 0), font=font)
+
+    pts1 = np.array(
+        [[0, 0], [box_width, 0], [box_width, box_height], [0, box_height]],
+        dtype=np.float32,
+    )
+    pts2 = np.array(box, dtype=np.float32)
+    M = cv2.getPerspectiveTransform(pts1, pts2)
+
+    img_right_text: np.ndarray = cv2.warpPerspective(
+        np.asarray(img_text, dtype=np.uint8),
+        M,
+        img_size,
+        flags=cv2.INTER_NEAREST,
+        borderMode=cv2.BORDER_CONSTANT,
+        borderValue=(255, 255, 255),
+    )
+    return img_right_text
+
+
+def _create_font(
+    txt: str,
+    size: Tuple[int, int],
+    font_path: str = "./landingai/fonts/default_font_ch_en.ttf",
+) -> ImageFont.FreeTypeFont:
+    font_size = int(size[1] * 0.99)
+    font = ImageFont.truetype(font_path, font_size, encoding="utf-8")
+    length = font.getsize(txt)[0]
+    if length > size[0]:
+        font_size = int(font_size * size[0] / length)
+        font = ImageFont.truetype(font_path, font_size, encoding="utf-8")
+    return font
```

Comment:
Too bad OpenCV does not yet support chinesse... Now that you have implemented this manually, does it make sense to use bbox_vizualizer for OD?? I remember you had issues controlling the font size

Reply:
good question, the issue of unable to dynamically change the font size is related to classification visualization. thought it could also apply to OD as well (not validated yet). 
so the `bbox_vizualizer` lib is still useful for drawing bbox and enabling customization for different bbox styles.

we can revisit this issue again later if we hear any feedback about the font size.